### ISSUE TITLE
Add dry_run param to auth function

### ISSUE THREAD

**Issue by F3real** [mozilla/mozilla_ci_tools#440]
Since auth was changed to be additional parameter for functions in buildapi_client, auth is required even  if dryRun is set to True. 
This means that client need to authenticate himself even if no request are going to be sent. Adding dryRun as additional param to auth would enable us to avoid this.
For backward compatibility we can add dryRun it so it defaults to False.

**Comment by armenzg** [comment]
I think we can just add dry_run to ever function which needs auth.
And then add "if dry_run" for every piece of code which would need to make use of auth.

#########################
Another idea, is that we could also introduce OO programing to buildapi_client.

The constructor takes auth and dry_run while most functions would loose auth and dry_run from their signature.
In the constructor we could validate various that auth/dry are incorrect:
* auth is not None and not dry_run --> raise exception

In other words, we instanciate a BuildapiClient(..., auth=something, dry_run={True,False}) and each function would be able to access self.auth and self.dry_run sa they need to.

**Comment by armenzg** [comment]
I think we should file the issue under buildapi_client as well.

**Comment by armenzg** [comment]
We can live without this.
